rudder-web/src/main/scala/bootstrap/liftweb/checks/CheckNcfTechniqueUpdate.scala

```diff
@@ -0,0 +1,150 @@
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package bootstrap.liftweb
+package checks
+
+import net.liftweb.common._
+import java.io.File
+import com.normation.eventlog.ModificationId
+import com.normation.utils.StringUuidGenerator
+import com.normation.rudder.domain.eventlog.RudderEventActor
+import com.normation.rudder.web.rest.RestExtractorService
+import com.normation.rudder.ncf.TechniqueWriter
+import scalaj.http.Http
+import net.liftweb.util.ControlHelpers.tryo
+import com.normation.rudder.api.ApiToken
+import monix.execution.Scheduler.{ global => scheduler }
+import scala.concurrent.duration._
+import com.normation.eventlog.EventActor
+import com.normation.rudder.api.ApiAccount
+import com.normation.rudder.ncf.ResultHelper.Result
+import com.normation.rudder.ncf.ResultHelper._
+import net.liftweb.json.JsonAST.JObject
+import net.liftweb.json.JsonAST.JArray
+import bootstrap.liftweb.BootstrapChecks
+
+/**
+ * Check at webapp startup if ncf Technique needs to be rewritten by Rudder
+ * Controlled by presence of flag file /opt/rudder/etc/force_ncf_technique_update
+ */
+class CheckNcfTechniqueUpdate(
+    restExtractor  : RestExtractorService
+  , techniqueWrite : TechniqueWriter
+  , systemApiToken : ApiAccount
+  , uuidGen        : StringUuidGenerator
+) extends BootstrapChecks {
+
+  override val description = "Regenerate all ncf techniques"
+
+  override def checks() : Unit = {
+
+    def request(path : String) = {
+      Http(s"http://localhost/ncf/api/${path}").
+        param("path", "/var/rudder/configuration-repository/ncf/").
+        header("X-API-TOKEN", systemApiToken.token.value)
+    }
+
+    val methodsRequest = request("generic_methods")
+    val techniquesRequest = request("techniques")
+
+    val ncfTechniqueUpdateFlag = "/opt/rudder/etc/force_ncf_technique_update"
+    val file =  new File(ncfTechniqueUpdateFlag)
+
+    def updateNcfTechniques : Unit = {
+      import net.liftweb.json.parse
+      import com.normation.utils.Control.sequence
+      ( for {
+        // First get generic methods
+        methodsResponse    <- tryo { methodsRequest.asString } ?~! "An error occurred while fetching generic methods from ncf API"
+        methodsResult      <- if(methodsResponse.isSuccess) {
+                                Full(methodsResponse.body)
+                              } else {
+                                scheduler.scheduleOnce(5.seconds)(updateNcfTechniques)
```

Comment:
I think you should not do that here, because it mixes the "when I do that" logic with the "what is done". 
And I'm not sure it is the only case for which we want to re schedule an inventory. 
So, what I propose: 
- add a:
  sealed trait NcfTechniqueUpgradeError (msg)
  object  NcfTechniqueUpgradeError { case class NcfApiRequest(msg), WriteTechnique(msg) }

- change Box => Either[NcfTechniqueUpgradeError, T]
- decide reschedule based on the kind of error.  (plus perhaps limit the number of retry ? Not sure it makes sense, though, because without the migration, it is just broken)
+ in the case of non recoverable error, make a big error message about what is broken, and what the user must do to make the situation work again.